homeassistant/components/vallox/__init__.py

```diff
@@ -0,0 +1,252 @@
+
+from homeassistant.const import CONF_HOST, CONF_NAME, CONF_SENSORS
+from homeassistant.helpers import discovery
+import homeassistant.helpers.config_validation as cv
+from homeassistant.helpers.discovery import async_load_platform
+from homeassistant.helpers.dispatcher import async_dispatcher_send
+from homeassistant.helpers.event import async_track_time_interval
+
+_LOGGER = logging.getLogger(__name__)
+
+DOMAIN = 'vallox'
+DEFAULT_NAME = 'Vallox'
+SIGNAL_VALLOX_STATE_UPDATE = "vallox_state_update"
+SCAN_INTERVAL = timedelta(seconds=60)
+
+# Various metric keys that are reused between profiles.
+METRIC_KEY_MODE = 'A_CYC_MODE'
+METRIC_KEY_PROFILE_FAN_SPEED_HOME = 'A_CYC_HOME_SPEED_SETTING'
+METRIC_KEY_PROFILE_FAN_SPEED_AWAY = 'A_CYC_AWAY_SPEED_SETTING'
+METRIC_KEY_PROFILE_FAN_SPEED_BOOST = 'A_CYC_BOOST_SPEED_SETTING'
+
+CONFIG_SCHEMA = vol.Schema({
+    DOMAIN: vol.Schema({
+        vol.Required(CONF_HOST): vol.All(ipaddress.ip_address, cv.string),
+        vol.Optional(CONF_NAME, default=DEFAULT_NAME): cv.string
+    }),
+}, extra=vol.ALLOW_EXTRA)
+
+# pylint: disable=no-member
+PROFILE_TO_STR_SETTABLE = {
+    VALLOX_PROFILE.HOME: 'Home',
+    VALLOX_PROFILE.AWAY: 'Away',
+    VALLOX_PROFILE.BOOST: 'Boost',
+    VALLOX_PROFILE.FIREPLACE: 'Fireplace',
+}
+
+STR_TO_PROFILE = {v: k for (k, v) in PROFILE_TO_STR_SETTABLE.items()}
+
+# pylint: disable=no-member
+PROFILE_TO_STR_REPORTABLE = {**{
+    VALLOX_PROFILE.NONE: 'None',
+    VALLOX_PROFILE.EXTRA: 'Extra',
+}, **PROFILE_TO_STR_SETTABLE}
+
+ATTR_PROFILE = 'profile'
+ATTR_PROFILE_FAN_SPEED = 'fan_speed'
+
+SERVICE_SCHEMA_SET_PROFILE = vol.Schema({
+    vol.Required(ATTR_PROFILE):
+        vol.All(cv.string, vol.In(STR_TO_PROFILE))
+})
+
+SERVICE_SCHEMA_SET_PROFILE_FAN_SPEED = vol.Schema({
+    vol.Required(ATTR_PROFILE_FAN_SPEED):
+        vol.All(vol.Coerce(int), vol.Clamp(min=0, max=100))
+})
+
+SERVICE_SET_PROFILE = 'set_profile'
+SERVICE_SET_PROFILE_FAN_SPEED_HOME = 'set_profile_fan_speed_home'
+SERVICE_SET_PROFILE_FAN_SPEED_AWAY = 'set_profile_fan_speed_away'
+SERVICE_SET_PROFILE_FAN_SPEED_BOOST = 'set_profile_fan_speed_boost'
+
+SERVICE_TO_METHOD = {
+        SERVICE_SET_PROFILE: {
+            'method': 'async_set_profile',
+            'schema': SERVICE_SCHEMA_SET_PROFILE},
+        SERVICE_SET_PROFILE_FAN_SPEED_HOME: {
+            'method': 'async_set_profile_fan_speed_home',
+            'schema': SERVICE_SCHEMA_SET_PROFILE_FAN_SPEED},
+        SERVICE_SET_PROFILE_FAN_SPEED_AWAY: {
+            'method': 'async_set_profile_fan_speed_away',
+            'schema': SERVICE_SCHEMA_SET_PROFILE_FAN_SPEED},
+        SERVICE_SET_PROFILE_FAN_SPEED_BOOST: {
+            'method': 'async_set_profile_fan_speed_boost',
+            'schema': SERVICE_SCHEMA_SET_PROFILE_FAN_SPEED},
+}
+
+DEFAULT_FAN_SPEED_HOME = 50
+DEFAULT_FAN_SPEED_AWAY = 25
+DEFAULT_FAN_SPEED_BOOST = 65
+
+
+async def async_setup(hass, config):
+    """Set up the client and boot the platforms."""
+    if DOMAIN not in config:
```

Comment:
This can't happen. There is no network discovery and we're not using config entries here yet.